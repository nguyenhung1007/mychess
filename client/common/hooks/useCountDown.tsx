import * as React from 'react';

export function useCountDown(totalTime: number, isStart: boolean, isPause: boolean): [number, boolean] {
    const [time, setTime] = React.useState<number>(totalTime);
    const [startTime] = React.useState(new Date());
    const [isFinish, setIsFinish] = React.useState(false);

    React.useEffect(() => {
        let intervalId: any;

        if (isStart && !isPause)
            intervalId = setInterval(() => {
                const passTime = new Date().getTime() - new Date(startTime).getTime();
                if (totalTime - passTime <= 0) setIsFinish(true);

                setTime(totalTime - passTime);
            }, 500);

        return () => clearInterval(intervalId);
    }, [time, isStart, isPause]);

    return [Math.round(time / 1000), isFinish];
}

export default useCountDown;
