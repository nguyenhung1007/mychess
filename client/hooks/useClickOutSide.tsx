import { useEffect, useRef } from 'react';

interface IUseClickOutSide {
        callBackOutside: Function;
        callbackInside?: Function;
        exceptElement?: string[];
}

export function useClickOutSide<T extends HTMLElement>({ callBackOutside, callbackInside, exceptElement = [] }: IUseClickOutSide) {
        const eleRef = useRef<T>(null);

        useEffect(() => {
                document.addEventListener('mousedown', handleOnClick, false);

                return () => {
                        document.removeEventListener('mousedown', handleOnClick, false);
                };
        }, []);

        const handleOnClick = (e: any) => {
                const isTriggerException = exceptElement.some((item) => e.target.classList.contains(item));
                if (isTriggerException) return;

                if (eleRef.current?.contains(e.target)) {
                        if (callbackInside) callbackInside();

                        return;
                }
                callBackOutside();
        };

        return eleRef;
}
