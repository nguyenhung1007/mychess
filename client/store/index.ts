import { combineReducers, configureStore } from '@reduxjs/toolkit';
import { apiReducer } from './api';
import { IApiState } from './api/interface';
import { authReducer } from './auth';
import { IAuthState } from './auth/interface';

export interface RootState {
        api: IApiState;
        auth: IAuthState;
}

const reducers = combineReducers<RootState>({
        api: apiReducer,
        auth: authReducer,
});

export const store = configureStore({
        reducer: reducers,
        devTools: process.env.NODE_ENV === 'production',
});
