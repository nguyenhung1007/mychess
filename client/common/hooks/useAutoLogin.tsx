import { useEffect } from 'react';
import { useSelector } from 'react-redux';
import Cookies from 'universal-cookie';
import userApi from '../../api/user';
import { RootState, store } from '../../store';
import { authActions } from '../../store/auth';
import { IAuthState } from '../../store/auth/interface';

export function useAutoLogin() {
    const cookies = new Cookies();
    const authState = useSelector<RootState, IAuthState>((state) => state.auth);

    useEffect(() => {
        const reToken = cookies.get('re-token');
        if (reToken) store.dispatch(authActions.updateLogin());
    }, []);

    useEffect(() => {
        if (authState.isLogin) store.dispatch(userApi.getLoginUser());
    }, [authState.isLogin]);
}

export default useAutoLogin;
