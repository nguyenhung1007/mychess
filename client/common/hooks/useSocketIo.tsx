import * as React from 'react';
import { useSelector } from 'react-redux';
import { RootState, store } from '../../store';
import { AuthState } from '../../store/auth/interface';
import * as socketIo from 'socket.io-client';
import authThunk from '../../store/auth/thunk';

interface UseSocketIoProps {
    namespace: string;
}

export const useSocketIo = ({ namespace }: UseSocketIoProps) => {
    const authState = useSelector<RootState, AuthState>((state) => state.auth);
    const ioClient = React.useRef<SocketIOClient.Socket>(socketIo.connect(`${process.env.SERVER_URL}/${namespace}`));

    React.useEffect(() => {
        if (authState.isSocketLogin) {
            ioClient.current.on('exception', (data: any) => {
                console.log(data);
            });
        } else store.dispatch(authThunk.getSocketToken());
    }, [authState.isSocketLogin]);
    return ioClient.current;
};

export default useSocketIo;
