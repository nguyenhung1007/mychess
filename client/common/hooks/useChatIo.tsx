import * as React from 'react';
import chatApi from '../../api/chatApi';
import { ServerResponse } from '../interface/api.interface';
import { Chat, ChatGatewayAction } from '../interface/chat.interface';
import useSocketIo from './useSocketIo';
import { useForm, UseFormRegister } from 'react-hook-form';
import { MessageDTO } from '../interface/dto/chat.dto';

const defaultValues: MessageDTO = {
    content: '',
};

export function useChatIo(chatId: string | undefined): [Chat | undefined, UseFormRegister<MessageDTO>, () => Promise<void>] {
    const clientChatIo = useSocketIo({ namespace: 'chat' });
    const { register, handleSubmit, reset } = useForm<MessageDTO>({ defaultValues });
    const [chat, setChat] = React.useState<Chat>();

    const handleOnSendMessage = (data: MessageDTO) => {
        if (chatId && data.content)
            chatApi.sendMessageChat({ chatId, content: data.content }).then(() => {
                reset(defaultValues);
            });
    };

    const onChatGet = (res: ServerResponse<Chat>) => setChat(res.data);
    const emitChatGet = () => clientChatIo.emit(ChatGatewayAction.CHAT_GET, { chatId });

    React.useEffect(() => {
        if (chatId) chatApi.joinChat({ chatId }).then(() => clientChatIo.emit(ChatGatewayAction.CHAT_JOIN, { chatId }));
    }, [chatId]);

    React.useEffect(() => {
        clientChatIo.on(ChatGatewayAction.CHAT_JOIN, emitChatGet);
        clientChatIo.on(ChatGatewayAction.CHAT_GET, onChatGet);

        return () => {
            clientChatIo.off(ChatGatewayAction.CHAT_JOIN, emitChatGet);
            clientChatIo.off(ChatGatewayAction.CHAT_GET, onChatGet);
        };
    }, [chatId]);

    return [chat, register, handleSubmit(handleOnSendMessage)];
}

export default useChatIo;
