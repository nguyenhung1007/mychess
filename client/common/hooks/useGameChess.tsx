import { useRouter } from 'next/router';
import * as React from 'react';
import { useSelector } from 'react-redux';
import { chessApi } from '../../api/chessApi';
import { RootState } from '../../store';
import { ServerResponse } from '../interface/api.interface';
import { ChessGatewayAction, ChessBoard, ChessPlayer, ChessStatus, ChessMoveRedis } from '../interface/chess.interface';
import { AuthState } from '../interface/user.interface';
import useSocketIo from './useSocketIo';
import routers from '../constants/router';

export function useGameChess(
    roomId: string,
): [
    ChessBoard | undefined,
    ChessPlayer[] | undefined,
    ChessMoveRedis[],
    React.RefObject<HTMLDivElement>,
    () => void,
    () => void,
    (x: number, y: number) => void,
    (x: number, y: number) => void,
    () => void,
] {
    const clientIoChess = useSocketIo({ namespace: 'chess' });
    const router = useRouter();
    const chessBoardRef = React.useRef<HTMLDivElement>(null);
    const [tttBoard, setTTTBoard] = React.useState<ChessBoard>();
    const [players, setPlayers] = React.useState<ChessPlayer[]>([]);
    const authState = useSelector<RootState, AuthState>((state) => state.auth);
    const [suggestion, setSuggestion] = React.useState<ChessMoveRedis[]>([]);

    const handleOnRestart = () => {
        if (tttBoard) {
            if (tttBoard.isBotMode)
                chessApi.createNewBotRoom().then((res) => {
                    const roomId = res.data.data.roomId;
                    router.push(`${routers.ticTacToePvP.link}/${roomId}`);
                });
            else chessApi.restartGame({ roomId });
        }
    };
    const handleOnStart = () => {
        chessApi
            .startGame({ roomId })
            .then(() => {
                if (chessBoardRef.current) {
                    const clientWidth = chessBoardRef.current.clientWidth / 2;
                    const scrollCenter = chessBoardRef.current.scrollWidth / 2 - clientWidth;
                    chessBoardRef.current.scrollLeft = scrollCenter;
                }
            })
            .catch(() => router.push(routers[404].link));
    };

    const handleOnAddMove = (x: number, y: number) => {
        const sound = new Audio('/asset/sounds/ttt-click.mp3');
        sound.volume = 0.5;
        sound.play();
        chessApi.addMovePvP({ roomId, x, y });
    };

    const handleOnSuggestion = (x: number, y: number) => {
        chessApi.getSuggestion({ roomId, x, y }).then((res) => {
            setSuggestion(res.data.data);
        });
    };

    const handleOnReady = () => {
        chessApi.readyGame({ roomId });
    };

    const emitTTTGet = () => clientIoChess.emit(ChessGatewayAction.CHESS_GET, { roomId });
    // const emitTTTCounter = () => clientIoChess.emit(ChessGatewayAction.TTT_COUNTER, { roomId });

    const onRestartGame = (res: ServerResponse<ChessBoard>) => router.push(`${routers.ticTacToePvP.link}/${res.data.id}`);
    const onTTTGet = (res: ServerResponse<ChessBoard>) => setTTTBoard(res.data);
    const onTTTCounter = (res: ServerResponse<ChessPlayer[]>) => setPlayers(res.data);

    // React.useEffect(() => {
    //     let interval: NodeJS.Timeout;

    //     if (tttBoard?.status === ChessStatus.PLAYING) {
    //         interval = setInterval(() => emitTTTCounter(), 1000);
    //     }

    //     return () => {
    //         clearInterval(interval);
    //     };
    // }, [tttBoard?.status, roomId]);

    React.useEffect(() => {
        if (tttBoard && tttBoard.status === ChessStatus.END) {
            const sound = new Audio('/asset/sounds/end-game.mp3');
            sound.volume = 0.5;
            sound.play();
        }
    }, [tttBoard?.status]);

    React.useEffect(() => {
        if (roomId && authState.isSocketLogin)
            chessApi
                .joinRoom({ roomId })
                .then(() => clientIoChess.emit(ChessGatewayAction.CHESS_JOIN, { roomId }))
                .catch(() => router.push(routers[404].link));
    }, [authState.isSocketLogin, roomId]);

    React.useEffect(() => {
        return () => {
            chessApi.leaveGame({ roomId });
        };
    }, [roomId]);

    React.useEffect(() => {
        clientIoChess.on(ChessGatewayAction.CHESS_GET, onTTTGet);
        clientIoChess.on(ChessGatewayAction.CHESS_JOIN, emitTTTGet);
        // clientIoChess.on(ChessGatewayAction.TTT_RESTART, onRestartGame);
        // clientIoChess.on(ChessGatewayAction.TTT_COUNTER, onTTTCounter);

        return () => {
            // clientIoChess.off(ChessGatewayAction.TTT_COUNTER, onTTTCounter);
            // clientIoChess.off(ChessGatewayAction.TTT_RESTART, onRestartGame);
            clientIoChess.off(ChessGatewayAction.CHESS_GET, onTTTGet);
            clientIoChess.off(ChessGatewayAction.CHESS_JOIN, emitTTTGet);
        };
    }, [roomId]);

    return [tttBoard, players, suggestion, chessBoardRef, handleOnReady, handleOnStart, handleOnAddMove, handleOnSuggestion, handleOnRestart];
}

export default useGameChess;
