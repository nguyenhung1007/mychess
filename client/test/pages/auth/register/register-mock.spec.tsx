import * as React from 'react';
import { fireEvent, render, waitFor } from '@testing-library/react';
import { Provider, useSelector } from 'react-redux';
import { useFormError } from '../../../../common/hooks/useFormError';
import Register from '../../../../pages/auth/register';
import { store } from '../../../../store';

jest.mock('../../../../common/hooks/useFormError');
jest.mock('react-redux', () => ({
    ...jest.requireActual('react-redux'),
    useSelector: jest.fn(),
}));

describe('Register Mock', () => {
    const mockFn = jest.fn();
    const useFormErrorMock = useFormError as jest.Mock;
    const useSelectorMock = useSelector as jest.Mock<any, any>;

    beforeEach(() => {
        useSelectorMock.mockReturnValue({
            isLoading: false,
            errorDetails: {},
            isError: false,
            message: '',
        });
        useFormErrorMock.mockReturnValue({ username: '', password: '', name: '', confirmPassword: '' });
    });

    afterEach(() => {
        useFormErrorMock.mockClear();
        useSelectorMock.mockClear();
    });

    it('Render Correct Default Mock', async () => {
        let wrapper = render(
            <Provider store={store}>
                <Register handleOnSubmit={mockFn} />
            </Provider>,
        );

        const nameTextField = await wrapper.findByTestId('name-text-field');
        const confirmPasswordTextField = await wrapper.findByTestId('confirmPassword-text-field');
        const usernameTextField = await wrapper.findByTestId('username-text-field');
        const passwordTextField = await wrapper.findByTestId('password-text-field');

        expect(usernameTextField).toBeDefined();
        expect(passwordTextField).toBeDefined();
        expect(nameTextField).toBeDefined();
        expect(confirmPasswordTextField).toBeDefined();
        expect(wrapper).toMatchSnapshot();
        try {
            await wrapper.findByTestId('username-error');
        } catch (err) {
            expect(err).toBeDefined();
        }

        try {
            await wrapper.findByTestId('wave-loading');
        } catch (err) {
            expect(err).toBeDefined();
        }
        try {
            await wrapper.findByTestId('success-msg');
        } catch (err) {
            expect(err).toBeDefined();
        }
    });

    it('Render with error label', async () => {
        useFormErrorMock.mockReturnValue({
            username: 'error-message',
            password: 'error-message',
            name: 'error-message',
            confirmPassword: 'error-message',
        });

        const wrapper = render(
            <Provider store={store}>
                <Register handleOnSubmit={mockFn} />
            </Provider>,
        );

        const nameError = await wrapper.findByTestId('name-error');
        const usernameError = await wrapper.findByTestId('username-error');
        const passwordError = await wrapper.findByTestId('password-error');
        const confirmPasswordError = await wrapper.findByTestId('confirmPassword-error');

        expect(nameError.innerHTML).toContain('Name error-message');
        expect(confirmPasswordError.innerHTML).toContain('Confirm Password error-message');
        expect(usernameError.innerHTML).toContain('Username error-message');
        expect(passwordError.innerHTML).toContain('Password error-message');
    });

    it('Render with loading ', async () => {
        useSelectorMock.mockReturnValue({ isLoading: true, errorDetails: {}, isError: false, message: '' });

        const wrapper = render(
            <Provider store={store}>
                <Register handleOnSubmit={mockFn} />
            </Provider>,
        );

        const loadingWave = await wrapper.findByTestId('wave-loading');
        expect(loadingWave).toBeDefined();
    });
    it('Render with success message ', async () => {
        useSelectorMock.mockReturnValue({ isLoading: false, errorDetails: {}, isError: false, message: 'hello' });

        const wrapper = render(
            <Provider store={store}>
                <Register handleOnSubmit={mockFn} />
            </Provider>,
        );

        const successMsg = await wrapper.findByText('hello');
        expect(successMsg).toBeDefined();
    });

    it('Function submit', async () => {
        const submitMockFn = jest.fn();

        const wrapper = render(
            <Provider store={store}>
                <Register handleOnSubmit={submitMockFn} />
            </Provider>,
        );
        const formLogin = await wrapper.findByTestId('register-form');

        waitFor(
            () => {
                fireEvent.submit(formLogin);
                expect(submitMockFn).toHaveBeenCalled();
            },
            { timeout: 1000 },
        );
    });
});
