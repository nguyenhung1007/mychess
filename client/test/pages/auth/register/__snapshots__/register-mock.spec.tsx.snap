// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Mock Render Correct Default Mock 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex-1 chess-bg grid place-items-center grid-rows-max shadow-sm"
      >
        <div
          class="bg-gray-800 px-4 md:px-10 py-16 w-full max-w-md rounded-sm fade-in "
        >
          <form
            data-testid="auth-register"
          >
            <h1
              class="text-center text-4xl text-white mb-7"
            >
              Register Account
            </h1>
            <div
              class="space-y-2"
            >
              <div
                class="space-y-1.5 text-sm"
              >
                <label
                  class="block text-cotton-seed "
                  data-testid="textfield-label-name"
                  for="name"
                >
                  Name
                </label>
                <input
                  autocomplete="on"
                  class="block w-full outline-none rounded-sm bg-tuna py-2 px-1.5  text-mercury"
                  data-testid="textfield-input-name"
                  id="name"
                  name="name"
                  type="text"
                />
              </div>
              <div
                class="space-y-1.5 text-sm"
              >
                <label
                  class="block text-cotton-seed "
                  data-testid="textfield-label-username"
                  for="username"
                >
                  Username
                </label>
                <input
                  autocomplete="on"
                  class="block w-full outline-none rounded-sm bg-tuna py-2 px-1.5  text-mercury"
                  data-testid="textfield-input-username"
                  id="username"
                  name="username"
                  type="text"
                />
              </div>
              <div
                class="space-y-1.5 text-sm"
              >
                <label
                  class="block text-cotton-seed "
                  data-testid="textfield-label-password"
                  for="password"
                >
                  Password
                </label>
                <input
                  autocomplete="off"
                  class="block w-full outline-none rounded-sm bg-tuna py-2 px-1.5  text-mercury"
                  data-testid="textfield-input-password"
                  id="password"
                  name="password"
                  type="password"
                />
              </div>
              <div
                class="space-y-1.5 text-sm"
              >
                <label
                  class="block text-cotton-seed "
                  data-testid="textfield-label-confirmpassword"
                  for="confirmPassword"
                >
                  Confirm Password
                </label>
                <input
                  autocomplete="off"
                  class="block w-full outline-none rounded-sm bg-tuna py-2 px-1.5  text-mercury"
                  data-testid="textfield-input-confirmpassword"
                  id="confirmPassword"
                  name="confirmPassword"
                  type="password"
                />
              </div>
            </div>
            <div
              class="mt-4 mb-7"
            >
              <div
                class="text-right"
                data-testid="sidelink-wrapper-sign-in-instead"
              >
                <a
                  class="text-mercury text-sm duration-300 hover:text-malibu"
                  data-testid="sidelink-link-sign-in-instead"
                  href="/auth/login"
                >
                  Sign In Instead
                </a>
              </div>
            </div>
            <button
              class="bg-btn-1 w-full py-2 text-white rounded-sm focus:outline-none "
              data-testid="btnform-sign-up"
              type="submit"
            >
              Sign Up
            </button>
          </form>
          <p
            class="text-center my-4 text-mercury"
          >
            Or continue with
          </p>
          <div
            class="space-y-4"
          >
            <div
              aria-hidden="true"
              class="bg-gray-700 py-2 px-4 flex space-x-4 text-mercury font-semibold rounded-sm duration-300 hover:bg-gray-600 cursor-pointer"
              data-testid="loginsocial-continue-with-google"
            >
              <svg
                fill="none"
                height="25"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  clip-path="url(#a)"
                >
                  <path
                    d="m5.319 15.003-.835 3.12-3.054.064A11.946 11.946 0 0 1 0 12.5c0-1.99.484-3.867 1.342-5.519l2.719.499 1.19 2.702a7.133 7.133 0 0 0-.384 2.318c0 .88.16 1.725.452 2.503z"
                    fill="#FBBB00"
                  />
                  <path
                    d="M23.79 10.258a12.028 12.028 0 0 1-.053 4.747 11.998 11.998 0 0 1-4.225 6.853l-3.424-.175-.485-3.025a7.152 7.152 0 0 0 3.077-3.652h-6.416v-4.748H23.79z"
                    fill="#518EF8"
                  />
                  <path
                    d="M19.512 21.858h.001a11.95 11.95 0 0 1-7.512 2.642c-4.57 0-8.544-2.554-10.57-6.313l3.888-3.183a7.135 7.135 0 0 0 10.285 3.654l3.908 3.2z"
                    fill="#28B446"
                  />
                  <path
                    d="m19.66 3.263-3.888 3.182a7.137 7.137 0 0 0-10.52 3.737l-3.91-3.2A11.998 11.998 0 0 1 12 .5a11.95 11.95 0 0 1 7.66 2.763z"
                    fill="#F14336"
                  />
                </g>
                <defs>
                  <clippath
                    id="a"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="#fff"
                      transform="translate(0 .5)"
                    />
                  </clippath>
                </defs>
              </svg>
              <span>
                Continue with Google
              </span>
            </div>
            <div
              aria-hidden="true"
              class="bg-gray-700 py-2 px-4 flex space-x-4 text-mercury font-semibold rounded-sm duration-300 hover:bg-gray-600 cursor-pointer"
              data-testid="loginsocial-continue-with-facebook"
            >
              <svg
                fill="none"
                height="25"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 24.5c6.627 0 12-5.373 12-12S18.627.5 12 .5 0 5.873 0 12.5s5.373 12 12 12z"
                  fill="#3B5998"
                />
                <path
                  d="M15.017 12.97h-2.141v7.844H9.632V12.97H8.089v-2.757h1.543V8.429c0-1.276.606-3.274 3.273-3.274l2.403.01v2.676h-1.744c-.286 0-.688.143-.688.752v1.622h2.425l-.284 2.755z"
                  fill="#fff"
                />
              </svg>
              <span>
                Continue with Facebook
              </span>
            </div>
            <div
              aria-hidden="true"
              class="bg-gray-700 py-2 px-4 flex space-x-4 text-mercury font-semibold rounded-sm duration-300 hover:bg-gray-600 cursor-pointer"
              data-testid="loginsocial-continue-with-github"
            >
              <svg
                fill="none"
                height="23"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 0c6.63 0 12 5.28 12 11.792 0 5.21-3.438 9.63-8.205 11.188-.6.11-.82-.254-.82-.567 0-.28.01-1.022.015-2.005 3.338.71 4.042-1.582 4.042-1.582.546-1.361 1.335-1.725 1.335-1.725 1.087-.731-.084-.716-.084-.716-1.205.082-1.838 1.215-1.838 1.215-1.07 1.803-2.809 1.282-3.495.98-.108-.762-.417-1.281-.76-1.576 2.665-.295 5.466-1.31 5.466-5.827 0-1.287-.465-2.34-1.235-3.164.135-.298.54-1.497-.105-3.121 0 0-1.005-.316-3.3 1.209-.96-.262-1.98-.392-3-.398-1.02.006-2.04.136-3 .398-2.28-1.525-3.285-1.21-3.285-1.21-.645 1.625-.24 2.824-.12 3.122-.765.825-1.23 1.877-1.23 3.164 0 4.53 2.805 5.527 5.475 5.817-.42.354-.81 1.077-.81 2.182 0 1.578.015 2.846.015 3.229 0 .309-.21.678-.825.56C3.435 21.417 0 16.995 0 11.792 0 5.28 5.373 0 12 0z"
                  fill="#fff"
                />
              </svg>
              <span>
                Continue with Github
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex-1 chess-bg grid place-items-center grid-rows-max shadow-sm"
    >
      <div
        class="bg-gray-800 px-4 md:px-10 py-16 w-full max-w-md rounded-sm fade-in "
      >
        <form
          data-testid="auth-register"
        >
          <h1
            class="text-center text-4xl text-white mb-7"
          >
            Register Account
          </h1>
          <div
            class="space-y-2"
          >
            <div
              class="space-y-1.5 text-sm"
            >
              <label
                class="block text-cotton-seed "
                data-testid="textfield-label-name"
                for="name"
              >
                Name
              </label>
              <input
                autocomplete="on"
                class="block w-full outline-none rounded-sm bg-tuna py-2 px-1.5  text-mercury"
                data-testid="textfield-input-name"
                id="name"
                name="name"
                type="text"
              />
            </div>
            <div
              class="space-y-1.5 text-sm"
            >
              <label
                class="block text-cotton-seed "
                data-testid="textfield-label-username"
                for="username"
              >
                Username
              </label>
              <input
                autocomplete="on"
                class="block w-full outline-none rounded-sm bg-tuna py-2 px-1.5  text-mercury"
                data-testid="textfield-input-username"
                id="username"
                name="username"
                type="text"
              />
            </div>
            <div
              class="space-y-1.5 text-sm"
            >
              <label
                class="block text-cotton-seed "
                data-testid="textfield-label-password"
                for="password"
              >
                Password
              </label>
              <input
                autocomplete="off"
                class="block w-full outline-none rounded-sm bg-tuna py-2 px-1.5  text-mercury"
                data-testid="textfield-input-password"
                id="password"
                name="password"
                type="password"
              />
            </div>
            <div
              class="space-y-1.5 text-sm"
            >
              <label
                class="block text-cotton-seed "
                data-testid="textfield-label-confirmpassword"
                for="confirmPassword"
              >
                Confirm Password
              </label>
              <input
                autocomplete="off"
                class="block w-full outline-none rounded-sm bg-tuna py-2 px-1.5  text-mercury"
                data-testid="textfield-input-confirmpassword"
                id="confirmPassword"
                name="confirmPassword"
                type="password"
              />
            </div>
          </div>
          <div
            class="mt-4 mb-7"
          >
            <div
              class="text-right"
              data-testid="sidelink-wrapper-sign-in-instead"
            >
              <a
                class="text-mercury text-sm duration-300 hover:text-malibu"
                data-testid="sidelink-link-sign-in-instead"
                href="/auth/login"
              >
                Sign In Instead
              </a>
            </div>
          </div>
          <button
            class="bg-btn-1 w-full py-2 text-white rounded-sm focus:outline-none "
            data-testid="btnform-sign-up"
            type="submit"
          >
            Sign Up
          </button>
        </form>
        <p
          class="text-center my-4 text-mercury"
        >
          Or continue with
        </p>
        <div
          class="space-y-4"
        >
          <div
            aria-hidden="true"
            class="bg-gray-700 py-2 px-4 flex space-x-4 text-mercury font-semibold rounded-sm duration-300 hover:bg-gray-600 cursor-pointer"
            data-testid="loginsocial-continue-with-google"
          >
            <svg
              fill="none"
              height="25"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#a)"
              >
                <path
                  d="m5.319 15.003-.835 3.12-3.054.064A11.946 11.946 0 0 1 0 12.5c0-1.99.484-3.867 1.342-5.519l2.719.499 1.19 2.702a7.133 7.133 0 0 0-.384 2.318c0 .88.16 1.725.452 2.503z"
                  fill="#FBBB00"
                />
                <path
                  d="M23.79 10.258a12.028 12.028 0 0 1-.053 4.747 11.998 11.998 0 0 1-4.225 6.853l-3.424-.175-.485-3.025a7.152 7.152 0 0 0 3.077-3.652h-6.416v-4.748H23.79z"
                  fill="#518EF8"
                />
                <path
                  d="M19.512 21.858h.001a11.95 11.95 0 0 1-7.512 2.642c-4.57 0-8.544-2.554-10.57-6.313l3.888-3.183a7.135 7.135 0 0 0 10.285 3.654l3.908 3.2z"
                  fill="#28B446"
                />
                <path
                  d="m19.66 3.263-3.888 3.182a7.137 7.137 0 0 0-10.52 3.737l-3.91-3.2A11.998 11.998 0 0 1 12 .5a11.95 11.95 0 0 1 7.66 2.763z"
                  fill="#F14336"
                />
              </g>
              <defs>
                <clippath
                  id="a"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="#fff"
                    transform="translate(0 .5)"
                  />
                </clippath>
              </defs>
            </svg>
            <span>
              Continue with Google
            </span>
          </div>
          <div
            aria-hidden="true"
            class="bg-gray-700 py-2 px-4 flex space-x-4 text-mercury font-semibold rounded-sm duration-300 hover:bg-gray-600 cursor-pointer"
            data-testid="loginsocial-continue-with-facebook"
          >
            <svg
              fill="none"
              height="25"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 24.5c6.627 0 12-5.373 12-12S18.627.5 12 .5 0 5.873 0 12.5s5.373 12 12 12z"
                fill="#3B5998"
              />
              <path
                d="M15.017 12.97h-2.141v7.844H9.632V12.97H8.089v-2.757h1.543V8.429c0-1.276.606-3.274 3.273-3.274l2.403.01v2.676h-1.744c-.286 0-.688.143-.688.752v1.622h2.425l-.284 2.755z"
                fill="#fff"
              />
            </svg>
            <span>
              Continue with Facebook
            </span>
          </div>
          <div
            aria-hidden="true"
            class="bg-gray-700 py-2 px-4 flex space-x-4 text-mercury font-semibold rounded-sm duration-300 hover:bg-gray-600 cursor-pointer"
            data-testid="loginsocial-continue-with-github"
          >
            <svg
              fill="none"
              height="23"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 0c6.63 0 12 5.28 12 11.792 0 5.21-3.438 9.63-8.205 11.188-.6.11-.82-.254-.82-.567 0-.28.01-1.022.015-2.005 3.338.71 4.042-1.582 4.042-1.582.546-1.361 1.335-1.725 1.335-1.725 1.087-.731-.084-.716-.084-.716-1.205.082-1.838 1.215-1.838 1.215-1.07 1.803-2.809 1.282-3.495.98-.108-.762-.417-1.281-.76-1.576 2.665-.295 5.466-1.31 5.466-5.827 0-1.287-.465-2.34-1.235-3.164.135-.298.54-1.497-.105-3.121 0 0-1.005-.316-3.3 1.209-.96-.262-1.98-.392-3-.398-1.02.006-2.04.136-3 .398-2.28-1.525-3.285-1.21-3.285-1.21-.645 1.625-.24 2.824-.12 3.122-.765.825-1.23 1.877-1.23 3.164 0 4.53 2.805 5.527 5.475 5.817-.42.354-.81 1.077-.81 2.182 0 1.578.015 2.846.015 3.229 0 .309-.21.678-.825.56C3.435 21.417 0 16.995 0 11.792 0 5.28 5.373 0 12 0z"
                fill="#fff"
              />
            </svg>
            <span>
              Continue with Github
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
